pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -133,7 +133,13 @@ public CompletableFuture<Void> deleteTopicPoliciesAsync(TopicName topicName) {
         if (NamespaceService.isHeartbeatNamespace(topicName.getNamespaceObject()) || isSelf(topicName)) {
             return CompletableFuture.completedFuture(null);
         }
-        return sendTopicPolicyEvent(topicName, ActionType.DELETE, null);
+        return pulsarService.getNamespaceService().checkTopicExists(NamespaceEventsSystemTopicFactory
+                .getEventsTopicName(topicName.getNamespaceObject())).thenCompose(topicExistsInfo -> {
+            if (topicExistsInfo.isExists()) {
+                return CompletableFuture.completedFuture(null);
+            }
+            return sendTopicPolicyEvent(topicName, ActionType.DELETE, null);
```

Comment:
please add a brief code comment about the reason for this logic.

Reply:
Added brief code comment. And I found a bug `if (topicExistsInfo.isExists()) -> if (!topicExistsInfo.isExists())` ❤️